### PR TITLE
Fix torrentday searches

### LINKED ISSUE #639: [APP SUBMITTED]: No JSON object could be decoded
### INFO
Python Version: **2.7.6 (default, Jun 22 2015, 18:00:18) [GCC 4.8.2]**
Operating System: **Linux-3.19.0-56-generic-i686-with-Ubuntu-14.04-trusty**
Locale: UTF-8
Branch: **develop**
Commit: PyMedusa/SickRage@3351698fd0d3b554f0dd6bbd8d0cbee0b60ce128
Link to Log: https://gist.github.com/0f9bcafda8153bec981bf232c004d72e
### ERROR
```
SEARCHQUEUE-MANUAL-311937 :: [TorrentDay] :: [3351698] Unknown exception while searching TorrentDay. Error: 'No JSON object could be decoded'
Traceback (most recent call last):
  File "/home/home/SickRage/sickbeard/search.py", line 547, in searchProviders
    forced_search, downCurQuality, manual_search, manual_search_type)
  File "/home/home/SickRage/sickrage/providers/GenericProvider.py", line 149, in find_search_results
    items_list += self.search(search_string, ep_obj=episode)
  File "/home/home/SickRage/sickbeard/providers/torrentday.py", line 123, in search
    parsedJSON = self.get_url(self.urls['search'], post_data=post_data, returns='json')
  File "/home/home/SickRage/sickrage/providers/GenericProvider.py", line 356, in get_url
    return getURL(url, post_data, params, self.headers, timeout, self.session, **kwargs)
  File "/home/home/SickRage/sickbeard/helpers.py", line 1472, in getURL
    return resp if response_type == u'response' or response_type is None else resp.json() if response_type == u'json' else getattr(resp, response_type, resp)
  File "/home/home/SickRage/lib/requests/models.py", line 808, in json
    return complexjson.loads(self.text, **kwargs)
  File "/home/home/SickRage/lib/simplejson/__init__.py", line 307, in loads
    return _default_decoder.decode(s)
  File "/home/home/SickRage/lib/simplejson/decoder.py", line 335, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/home/SickRage/lib/simplejson/decoder.py", line 353, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```
---
_STAFF NOTIFIED_: @pymedusa/support @pymedusa/moderators

### PR DESCRIPTION
**Pull request by labrys**
- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

Fixes #639 
Check provider response status and log errors
Check that provider response is actually json

### CODE REVIEW

File: sickbeard/providers/torrentday.py [hunk 1]
```diff
@@ -120,19 +121,25 @@ def search(self, search_params, age=0, ep_obj=None):  # pylint: disable=too-many
...
+                except RequestException as msg:
+                    logger.log(u'Error while connecting to provider: {error}'.format(error=msg), logger.ERROR)
                     continue
 
                 try:
-                    torrents = parsedJSON.get('Fs', [])[0].get('Cn', {}).get('torrents', [])
-                except Exception:
+                    jdata = response.json()
+                except ValueError:  # also catches JSONDecodeError if simplejson is installed
+                    logger.log(u"Data returned from provider is not json", logger.ERROR)
```
Comment (labrys) [inline sickbeard/providers/torrentday.py]: Making these logs `ERROR` for now.  Will likely downgrade in the future.

**Comment by fernandog** [comment]
problem is that when site is down or a plain text response like 'maintaince' this happens. 
need to find the response text to proper fix

@duramato

**Comment by labrys** [comment]
@fernandog Agreed, this will (for now at least) fail clean without a traceback though, as a traceback for this issue isn't necessary.  That's why I opted to get the response (so we can parse it in the future) and a log level of error for now (to continue to see the frequency this occurs, although warning could also suffice),

**Comment by labrys** [comment]
@duramato If you don't mind I'd like you to run a branch that will save the response output when this occurs so we can confirm the information that needs to be parsed.

**Comment by labrys** [comment]
Approved

**Pull request merged**